libraries/chain/block_log.cpp

```diff
@@ -1023,7 +1028,7 @@ namespace eosio { namespace chain {
          }
 
          // read in the previous contiguous memory into the read buffer
```

Comment:
Isn't the code within this for loop going to write each buffer in the wrong order? It seems like it would be fine if `to_write <= detail::reverse_iterator::_buf_len`. But if it is larger, then the later sections in the original block log would be written out in the `new_block_file` before the earlier sections. Or am I missing something?

Reply:
read_buffer = to_write

Reply:
also, wasn't writing to the new file in the correct location, fixed that and the mode the new cfile was opened in.